### PR TITLE
scan: Implement Sub and wire up for Average Block time and Inflation rate

### LINKED ISSUE #1146: Implement Sub and wire up for Average Block time and Inflation rate

### PR DESCRIPTION
**Pull request by tansawit**
Fixes: #1146

### CODE REVIEW

File: scan/src/pages/ValidatorHomePage.re [hunk 1]
```diff
@@ -187,16 +187,39 @@ let renderBody = (rank, validator: ValidatorSub.t) => {
...
+      );
+    React.useEffect0(() => {
+      let newPrevDay = MomentRe.momentNow() |> MomentRe.Moment.toUnix |> float_of_int;
+      let timeOutId = Js.Global.setTimeout(() => setPrevDay(_ => newPrevDay), 1000);
+      Some(() => Js.Global.clearTimeout(timeOutId));
+    });
+
     let pageSize = 10;
 
+    let lastDay = prevDay |> Js.Json.number;
```
Comment (evilpeach) [inline scan/src/pages/ValidatorHomePage.re]: No need to declare a new variable. Just use `BlockSub.pastDayCount(prevDay |> Js.Json.number);`

File: scan/src/pages/ValidatorHomePage.re [hunk 2]
```diff
@@ -187,16 +187,39 @@ let renderBody = (rank, validator: ValidatorSub.t) => {
...
+        (
+          MomentRe.momentNow()
+          |> MomentRe.Moment.subtract(~duration=MomentRe.duration(1., `days))
+          |> MomentRe.Moment.toUnix
+          |> float_of_int
+        )
+        *. 1000.
+      );
+    React.useEffect0(() => {
+      let newPrevDay = MomentRe.momentNow() |> MomentRe.Moment.toUnix |> float_of_int;
```
Comment (evilpeach) [inline scan/src/pages/ValidatorHomePage.re]: Is it correct? Did you forget to subtract the duration?

**Review by evilpeach — CHANGES_REQUESTED** [review]

File: scan/src/pages/ValidatorHomePage.re [hunk 3]
```diff
@@ -187,16 +187,44 @@ let renderBody = (rank, validator: ValidatorSub.t) => {
...
+    React.useEffect0(() => {
+      let newPrevDay =
+        (
+          MomentRe.momentNow()
+          |> MomentRe.Moment.subtract(~duration=MomentRe.duration(1., `days))
+          |> MomentRe.Moment.toUnix
+          |> float_of_int
+        )
+        *. 1000.;
+      let timeOutId = Js.Global.setTimeout(() => setPrevDay(_ => newPrevDay), 60_000);
```
Comment (prin-r) [inline scan/src/pages/ValidatorHomePage.re]: Use setInterval instead.

**Review by prin-r — APPROVED** [review]

File: scan/src/pages/ValidatorHomePage.re [hunk 4]
```diff
@@ -187,16 +187,44 @@ let renderBody = (rank, validator: ValidatorSub.t) => {
...
+          |> MomentRe.Moment.subtract(~duration=MomentRe.duration(1., `days))
+          |> MomentRe.Moment.toUnix
+          |> float_of_int
+        )
+        *. 1000.
+      );
+    React.useEffect0(() => {
+      let newPrevDay =
+        (
+          MomentRe.momentNow()
```
Comment (sorawit) [inline scan/src/pages/ValidatorHomePage.re]: Dont write the same code twice if not necessary. DRY.

**Review by prin-r — CHANGES_REQUESTED** [review]
-

File: scan/src/pages/ValidatorHomePage.re [hunk 5]
```diff
@@ -205,7 +233,10 @@ let make = () =>
     let allValidatorCount =
       bondedValidatorCount + unbondedValidatorCount + unbondingValidatorCount;
 
-    let allBondedAmount = bondedValidatorCount;
+    //TODO: Replace with real value
+    let allBondedAmount = 400;
```
Comment (sorawit) [inline scan/src/pages/ValidatorHomePage.re]: We have this data already, let's not mock it?

**Review by sorawit — CHANGES_REQUESTED** [review]

File: scan/src/pages/ValidatorHomePage.re (hunk 5 above)
Comment (tansawit) [reply scan/src/pages/ValidatorHomePage.re]: Will be fixedin in #1191 krub

**Comment by evilpeach** [comment]
@tansawit Please resolve the conflict.

**Review by evilpeach — DISMISSED** [review]
LGTM

**Comment by evilpeach** [comment]
@tansawit Please fix the conflict

File: scan/src/pages/ValidatorHomePage.re [hunk 6]
```diff
@@ -171,40 +126,115 @@ let renderBody = (rank, validator: ValidatorSub.t, bondedTokenCount) => {
...
 [@react.component]
 let make = () =>
   {
     let (page, setPage) = React.useState(_ => 1);
+
+    let (prevDayTime, setPrevDayTime) = React.useState(getPrevDay);
+    let (currentTime, setCurrentTime) = React.useState(getCurrentDay);
+
+    React.useEffect0(() => {
+      let timeOutId = Js.Global.setInterval(() => setPrevDayTime(getPrevDay), 60_000);
```
Comment (sorawit) [inline scan/src/pages/ValidatorHomePage.re]: Why don't we do it in the same set interval.

File: scan/src/pages/ValidatorHomePage.re [hunk 7]
```diff
@@ -89,80 +89,35 @@ let renderBody = (rank, validator: ValidatorSub.t, bondedTokenCount) => {
   let token = validator.tokens;
   let commission = validator.commission;
   let uptime = validator.nodeStatus.uptime;
-  // let allRequestCount =
-  //   validator.completedRequestCount + validator.missedRequestCount |> float_of_int;
-  // let reportRate = (validator.completedRequestCount |> float_of_int) /. allRequestCount *. 100.;
 
   <TBody key={rank |> string_of_int}>
```
Comment (sorawit) [inline scan/src/pages/ValidatorHomePage.re]: Do not use rank as key

**Review by sorawit — CHANGES_REQUESTED** [review]

**Review by sorawit — APPROVED** [review]

**Review by evilpeach — APPROVED** [review]

**Review by prin-r — APPROVED** [review]

**Pull request merged**